common/config/dynamic.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/odigos-io/odigos/common"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	DestinationName   = "DYNAMIC_DESTINATION_NAME"
+	DestinationType   = "DYNAMIC_DESTINATION_TYPE"
+	ConfigurationData = "DYNAMIC_CONFIGURATION_DATA"
```

Comment:
In Golang, anything that starts with a CAPITAL letter becomes "globally" accessible (e.g. use in files outside of this package).

These variable names are too generic to be global, consider using lower case 1st letter, or prefixing with "dynamic":

```suggestion
	destinationName   = "DYNAMIC_DESTINATION_NAME"
	destinationType   = "DYNAMIC_DESTINATION_TYPE"
	configurationData = "DYNAMIC_CONFIGURATION_DATA"
```

```suggestion
	DynamicDestinationName   = "DYNAMIC_DESTINATION_NAME"
	DynamicDestinationType   = "DYNAMIC_DESTINATION_TYPE"
	DynamicConfigurationData = "DYNAMIC_CONFIGURATION_DATA"
```